rrrspec-client/lib/rrrspec/client/rspec_runner_factory.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Client
+    class RSpecRunnerFactory
+      attr_reader :next_factory
+
+      def initialize(successor)
+        @next_factory = successor
+      end
+
+      def create(file_extension)
+        unless Thread.current[:rspec3_runner]
+          Thread.current[:rspec3_runner] = RSpecRunner.new
```

Comment:
Was having a hard time figuring out the process initialization and talking about with helps me.

I got it now. Was missing in my mental model the part that the worker spans the slave, that's why they run as processes and there are no concurrency problem. Inside the slave process there are only 2 thread, the pacemaker and the slave.